controllers/clusteroperator_controller_test.go

```diff
@@ -365,84 +365,104 @@ var _ = Describe("Component sync controller", func() {
-    "cloudControllerManagerOpenStack": "registry.ci.openshift.org/openshift:openstack-cloud-controller-manager"
-}`,
-		expectConfig: operatorConfig{
-			ControllerImage:  "registry.ci.openshift.org/openshift:openstack-cloud-controller-manager",
-			ManagedNamespace: defaultManagementNamespace,
-		},
-	}, {
-		name:      "Unmarshal images from file for unknown platform returns nothing",
-		namespace: "otherNamespace",
-		platform:  configv1.NonePlatformType,
-		imagesContent: `{
-    "cloudControllerManagerAWS": "registry.ci.openshift.org/openshift:aws-cloud-controller-manager",
-    "cloudControllerManagerOpenStack": "registry.ci.openshift.org/openshift:openstack-cloud-controller-manager"
-}`,
-		expectConfig: operatorConfig{
-			ControllerImage:  "",
-			ManagedNamespace: "otherNamespace",
-		},
-	}, {
-		name: "Broken JSON is rejected",
-		imagesContent: `{
-    "cloudControllerManagerAWS": BAD,
-}`,
-		expectError: true,
-	}}
-
-	for _, tc := range tc {
-		t.Run(tc.name, func(t *testing.T) {
-			file, err := ioutil.TempFile(os.TempDir(), "images")
-			path := file.Name()
-			if err != nil {
-				t.Fatal(err)
-			}
-			defer file.Close()
+var _ = Describe("Apply resources should", func() {
+	var resources []client.Object
+	var reconciler *CloudOperatorReconciler
+	var recorder *record.FakeRecorder
 
-			_, err = file.WriteString(tc.imagesContent)
-			if err != nil {
-				t.Fatal(err)
-			}
+	BeforeEach(func() {
+		c, err := cache.New(cfg, cache.Options{})
+		Expect(err).To(Succeed())
+		w, err := NewObjectWatcher(WatcherOptions{Cache: c})
+		Expect(err).To(Succeed())
 
-			r := &CloudOperatorReconciler{
-				ImagesFile:       path,
-				ManagedNamespace: tc.namespace,
-			}
-			config, err := r.composeConfig(tc.platform)
-			if isErr := err != nil; isErr != tc.expectError {
-				t.Fatalf("Unexpected error result: %v", err)
-			}
+		ns := &corev1.Namespace{}
+		ns.SetName("cluster-cloud-controller-manager")
+
+		resources = []client.Object{}
+		if !apierrors.IsNotFound(cl.Get(context.TODO(), client.ObjectKeyFromObject(ns), ns.DeepCopy())) {
+			Expect(cl.Create(context.TODO(), ns.DeepCopy())).ShouldNot(HaveOccurred())
+		}
+
+		recorder = record.NewFakeRecorder(32)
+		reconciler = &CloudOperatorReconciler{
+			Client:   cl,
+			Scheme:   scheme.Scheme,
+			Recorder: recorder,
+			watcher:  w,
+		}
 
-			if !equality.Semantic.DeepEqual(config, tc.expectConfig) {
-				t.Errorf("Config is not equal:\n%v\nexpected\n%v", config, tc.expectConfig)
+	})
+
+	It("Expect update when resources are not found", func() {
+		resources = append(resources, cloud.GetAWSResources()...)
+
+		updated, err := reconciler.applyResources(context.TODO(), resources)
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(updated).To(BeTrue())
+		Eventually(recorder.Events).Should(Receive(ContainSubstring("Resource was successfully updated")))
+	})
+
+	It("Expect update when deployment generation have changed", func() {
+		dep := &appsv1.Deployment{}
```

Comment:
Nit: To make it clearer that we are creating a reference to the deployment resource, I would init this in a different way
```suggestion
		var dep *appsv1.Deployment
```

With the current way it's like we are creating an empty one then starting to fill it out